### PR TITLE
test: add E2E tests for custom component health-check

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Proposed changes

Add E2E tests for health-check package

## Testing

- [x] Unit Tests:

- [ ] Functionnal Tests:

- [ ] Manual Tests:

### CODE REVIEW

File: packages/ui/atomic/create-atomic-component/template/src/components/sample-component/package.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "@coveo/sample-component",
   "version": "0.0.1",
-  "description": "@coveo/atomic component starter",
```
Comment (y-lakhdar) [inline packages/ui/atomic/create-atomic-component/template/src/components/sample-component/package.json:4]: forgot to remove the description from the template (part of CDX-1323)

**Review by olamothe — COMMENTED** [review]
Do we have any test for the non-happy path ?

Verifying that it blocks the publishing if the health check are not correctly respected ?

**Comment by y-lakhdar** [comment]
> Do we have any test for the non-happy path ?
> Verifying that it blocks the publishing if the health check are not correctly respected ?

Yes, non-happy paths are all listed and covered in packages/ui/atomic/health-check/src/index.spec.ts

File: packages/ui/atomic/create-atomic-result-component/tests/test.spec.ts [hunk 2]
```diff
@@ -108,5 +111,45 @@ describe(PACKAGE_NAME, () => {
         }).status
       ).toBe(0);
     });
+
+    describe('health-check', () => {
```
Comment (mrrajamanickam-coveo) [inline packages/ui/atomic/create-atomic-result-component/tests/test.spec.ts:115]: would it make sense to extract the common code to a test utils or use [table driven tests](https://jestjs.io/docs/api#describeeachtablename-fn-timeout) with parameterized pkg names?

Comment (mrrajamanickam-coveo) [reply packages/ui/atomic/create-atomic-result-component/tests/test.spec.ts:115]: > What part of the common code do you have in mind @mrrajamanickam-coveo?

most of the test code between the 2 pkgs seems the same - hence wondering about reducing duplication

Comment (louis-bompart) [reply packages/ui/atomic/create-atomic-result-component/tests/test.spec.ts:115]: Everything that does not contain a 'jest' keyword is refactorable IIRC

@y-lakhdar that'd be a big ask, but maybe it'd be interesting to compare side-by-side having all the e2e for the health-check package in the health-check package?

File: packages/ui/atomic/health-check/e2e/test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      testDirname: 'valid-arg',
+      packageName: 'valid-component-name',
+    },
+    {
+      describeName: 'when called with a component name without hyphen',
+      args: ['nohypen'],
+      testDirname: 'invalid-arg',
+      packageName: 'atomic-nohypen',
+    },
+  ])('$describeName', ({args, testDirname, packageName}) => {
```
Comment (louis-bompart) [inline packages/ui/atomic/health-check/e2e/test.spec.ts:62]: I think you do not need to test all the case here for the `health-check` package, it is  redundant.

Comment (y-lakhdar) [reply packages/ui/atomic/health-check/e2e/test.spec.ts:62]: Sorry, I forgot to move the PR to draft while I was refactoring E2E tests.
I was actually removing redundant tests :P

File: packages/ui/atomic/health-check/tsconfig.json [hunk 4]
```diff
@@ -6,8 +6,8 @@
     "module": "Node16",
     "target": "ESNext",
     "moduleResolution": "Node16",
-    "sourceMap": false,
-    "declaration": false,
+    "sourceMap": true,
+    "declaration": true,
```
Comment (y-lakhdar) [inline packages/ui/atomic/health-check/tsconfig.json:10]: Turn on declaration files to run the E2E tests. But don't include in the build that will be published

File: packages/ui/atomic/health-check/src/e2e/test.spec.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+      // TODO: decomment once tested are done with windows
+      // it('should be able to pass health checks', () => {
+      //   const {stdout} = publish(pkgName, testDirectory);
+      //   const message = stripAnsi(stdout.toString()).replace(/\n/g, '');
+      //   expect(message).not.toContain('✖');
+      //   expect(message).toContain('✔ Readme file');
+      //   expect(message).toContain('✔ Component name');
+      //   expect(message).toContain('✔ Required properties in package.json');
+      // });
```
Comment (louis-bompart) [inline packages/ui/atomic/health-check/src/e2e/test.spec.ts:152]: - [ ]  todo ;)

**Review by louis-bompart — CHANGES_REQUESTED** [review]
lgtm, when todo is done :D

File: packages/ui/atomic/health-check/src/e2e/test.spec.ts (hunk 5 above)
Comment (y-lakhdar) [reply packages/ui/atomic/health-check/src/e2e/test.spec.ts:152]: oups thanks

**Review by louis-bompart — APPROVED** [review]
:shipit:

**Pull request merged**